internal/charmstore/migrations.go

```diff
@@ -181,7 +184,35 @@ func populateReadACL(db StoreDatabase) error {
 		if err := baseEntities.UpdateId(entity.URL, bson.D{{
 			"$set", bson.D{{"acls.read", readPerm}},
 		}}); err != nil {
-			return errgo.Notef(err, "cannot populate ACL for base entity %s", entity.URL)
+			return errgo.Notef(err, "cannot populate read ACL for base entity %s", entity.URL)
+		}
+		counter++
+	}
+	if err := iter.Close(); err != nil {
+		return errgo.Notef(err, "cannot iterate base entities")
+	}
+	logger.Infof("%d base entities updated", counter)
+	return nil
+}
+
+// populateWriteACL adds the write ACL to base entities not having the field.
+func populateWriteACL(db StoreDatabase) error {
+	baseEntities := db.BaseEntities()
+	var entity mongodoc.BaseEntity
+	iter := baseEntities.Find(bson.D{{
+		"acls.write", bson.D{{"$exists", false}},
+	}, {
+		"user", bson.D{{"$ne", ""}},
+	}}).Select(bson.D{{"_id", 1}}).Iter()
+
+	defer iter.Close()
```

Comment:
Maybe log possible error from Close() like in the above populateReadACL function?